src/main/java/org/openrewrite/staticanalysis/MoveFieldsToTopOfClass.java

```diff
@@ -0,0 +1,136 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.Statement;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+@EqualsAndHashCode(callSuper = false)
+@Value
+public class MoveFieldsToTopOfClass extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Move fields to the top of class definition";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Reorders class members so that all field declarations appear before any method declarations, " +
+               "constructors, or other class members. This improves code organization and readability by " +
+               "grouping field declarations together at the top of the class. Comments associated with fields " +
+               "are preserved during the reordering.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, ctx);
+
+                List<Statement> statements = cd.getBody().getStatements();
+                if (statements.isEmpty()) {
+                    return cd;
+                }
+
+                List<Statement> fieldDeclarations = new ArrayList<>();
+                List<Statement> otherStatements = new ArrayList<>();
+
+                // Separate field declarations from other statements
+                for (Statement statement : statements) {
+                    if (statement instanceof J.VariableDeclarations) {
+                        fieldDeclarations.add(statement);
+                    } else {
+                        otherStatements.add(statement);
+                    }
+                }
+
+                // If all fields are already at the top, no changes needed
+                if (fieldDeclarations.isEmpty() || isAlreadyOrdered(statements, fieldDeclarations)) {
```

Comment:
The same is checked in the method called.
```suggestion
                if (isAlreadyOrdered(statements, fieldDeclarations)) {
```